File: C++/27.Remove Element.cpp
#include<iostream>
#include<vector>
using namespace std;
/*
27.删除元素
就当前数组删除元素且返回之后的数组长度
双指针思路返回数组长度正确但实际并未改变数组
*/
class Solution
{
private:
public:
    Solution();
    ~Solution();
    int removeElement(vector<int>& nums,int val)
    {
        int i=0;
        for (int j=0; j< nums.size(); j++){
            if(nums[j]!=val){
                //易错点 双指针中慢指针什么时候自增
                nums[i]=nums[j];
                i++;
            }
        }
        return i;
    }
};
Solution::Solution(){}
Solution::~Solution(){}
int main(int argc, char* argv[]){
    Solution* solution = new Solution();
    vector<int> nums({1,2,3,3,4,4,5});
    int val=4;
    cout<<solution->removeElement(nums,val)<<endl;
}

File: Python/27.Remove Element.py
#!/usr/bin/python3.7
"""
26.移除元素
从数组中删除元素
"""
class Solution(object):
    def removeElement(self, nums, val:int) -> int:
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        for i in range(len(nums))[::-1]:
            while val in nums:
                nums.remove(val)
        return len(nums)

def main():
    n = [1,2,3,4,4,5,5,5,6,6,7,7,7]
    val = 7
    a = Solution()
    print(Solution.removeElement(a,n,val))

if __name__ == '__main__':
    main()
